src/shapes/polyline.class.ts

```diff
@@ -79,56 +93,111 @@ import { makeBoundingBoxFromPoints } from '../util/misc/boundingBoxFromPoints';
-       * returning an object with left, top, width, height to measure the
-       * polygon size
-       * @return {Object} object.left X coordinate of the polygon leftmost point
-       * @return {Object} object.top Y coordinate of the polygon topmost point
-       * @return {Object} object.width distance between X coordinates of the polygon leftmost and rightmost point
-       * @return {Object} object.height distance between Y coordinates of the polygon topmost and bottommost point
+       * @returns {Point} top left position of the bounding box, useful for complementary positioning
+       */
+      setDimensions: function () {
+        const { left, top, width, height, pathOffset, strokeOffset } =
+          this._calcDimensions();
+        this.set({ width, height, pathOffset, strokeOffset });
+        return new Point(left, top);
+      },
+
+      /**
+       * @override stroke is taken in account in size
+       */
+      _getNonTransformedDimensions: function () {
+        return this.exactBoundingBox
+          ? new Point(this.width, this.height)
+          : this.callSuper('_getNonTransformedDimensions');
+      },
+
+      /**
+       * @override stroke and skewing are taken into account when projecting stroke on points,
+       * therefore we don't want the default calculation to account for skewing as well
+       *
        * @private
        */
-      _calcDimensions: function () {
-        const points = this.exactBoundingBox
-          ? this._projectStrokeOnPoints()
-          : this.points.map((p) => new Point(p));
-        if (points.length === 0) {
-          return makeBoundingBoxFromPoints([new Point(0, 0)]);
+      _getTransformedDimensions: function (options) {
+        return this.exactBoundingBox
+          ? this.callSuper('_getTransformedDimensions', {
+              ...(options || {}),
+              // disable stroke bbox calculations
+              strokeWidth: 0,
+              // disable skewing bbox calculations
+              skewX: 0,
+              skewY: 0,
+            })
+          : this.callSuper('_getTransformedDimensions', options);
+      },
+
+      /**
+       * Recalculates dimensions when changing skew and scale
+       * @private
+       */
+      _set: function (key, value) {
+        const changed = this.initialized && this[key] !== value;
+        const output = this.callSuper('_set', key, value);
+        if (
+          changed &&
+          (((key === 'scaleX' || key === 'scaleY') &&
+            this.strokeUniform &&
+            this.strokeLineJoin !== 'round') ||
+            // TODO: check if you really need to recalculate for all cases
+            key === 'skewX' ||
+            key === 'skewY' ||
+            key === 'strokeMiterLimit' ||
+            key === 'strokeLineCap' ||
+            key === 'strokeLineJoin' ||
+            key === 'strokeLineWidth' ||
+            key === 'strokeUniform' ||
+            key === 'points')
+        ) {
+          this.setDimensions();
```

Comment:
this wasn't automatic before even with the simple calculation, because the simple logic was good ( well except wasn't exact ) for every case.
So all good to me, but adding `points` here adds a new automatic feature and is partial.

I can set the whole array of points and i can get a bbox update, while i can't mutate a point and do the same.
Not sure this checking for points is a good idea outside skewX and skewY.
a end user can change skewX and skewY alone and so yes, maybe the bbox should update in that case, but the other changes are a developer choice, with JS code, and in that js code there should be a setDimensions too.

Reply:
We discussed it as well, the mutation.
Let's revert points.
`setDimensions` is exposed for the dev

Reply:
either you revert everything and leave skewX and skewY, or you leave as it is.
I do really prefer only skewX and skewY that are in user hands.

The rest is not forbidden, but is a different feature, and can be introduced with some sort of planning.
Text has something similar right?

propertiesThatChangesWidthHeightInText = [...array of stuff]

same we do with properties that invalidate caching

I'm not saying is wrong, but i would like to introduce it on its own, with a dedicated moment of planning

Reply:
> either you revert everything and leave skewX and skewY, or you leave as it is.

Not good enough accounting only for skew. I will introduce an array of props.

Reply:
301276fe

Reply:
i see the planning part wasn't ok.